packages/icon/package.py

```diff
@@ -0,0 +1,247 @@
+    depends_on('eccodes@2.18.0 +build_shared_libs', when='+eccodes', type=('build', 'link', 'run'))
+    depends_on('hdf5', type=('build', 'link', 'run'))
+    depends_on('claw@2.0.1', when='+claw', type=('build', 'link', 'run'))
+    depends_on('netcdf-c ~mpi', type=('build', 'link', 'run'))
+    depends_on('mpicuda', when='icon_target=gpu', type=('build', 'link', 'run'))
+    depends_on('ompt-openmp', when='%pgi', type=('build', 'link', 'run'))
+    depends_on('mpi', when='icon_target=cpu', type=('build', 'link', 'run'))
+    depends_on('cuda%gcc', when='icon_target=gpu', type=('build', 'link', 'run'))
+
+    variant('icon_target', default='gpu', description='Build with target gpu or cpu', values=('gpu', 'cpu'), multi=False)
+    variant('host', default='daint', description='Build on host daint', multi=False)
+    variant('cuda_arch', default='60', description='Build with cuda_arch', values=('70', '60', '37'), multi=False)
+    variant('claw', default=True, description='Build with claw directories enabled')
+    variant('rte-rrtmgp', default=True, description='Build with rte-rrtmgp enabled')
+    variant('mpi-checks', default=False, description='Build with mpi-check enabled')
+    variant('openmp', default=True, description='Build with openmp enabled')
+    variant('serialize', default=False, description='Build with serialization enabled')
+    variant('eccodes', default=False, description='Build with grib2 enabled')
+    variant('test_name', default='none', description='Launch test: test_name after installation')
+    variant('skip-config', default=False, description='Skip configure phase')
+    variant('ham', default=True, description='Build with hammoz and atm_phy_echam enabled.')
+
+    conflicts('+openmp', when='%intel')
+    conflicts('+openmp', when='%cce')
+    conflicts('+claw', when='%intel')
+    conflicts('+claw', when='%cce')
+    conflicts('icon_target=gpu', when='%intel')
+    conflicts('icon_target=gpu', when='%cce')
+
+    atm_phy_echam_submodels_namelists_dir = 'externals/atm_phy_echam_submodels/namelists'
+
+    def setup_build_environment(self, env):
+        self.setup_run_environment(env)
+        if self.spec.variants['icon_target'].value == 'gpu':
+            env.set('CUDA_HOME', self.spec['cuda'].prefix)
+
+    @run_before('configure')
+    def generate_hammoz_nml(self):
+        if '@ham' in self.spec:
+            with working_dir(self.atm_phy_echam_submodels_namelists_dir):
+                make()
+
+    def configure_args(self):
+        args = []
+
+        CFLAGS=''
+        CPPFLAGS=''
+        FCFLAGS=''
+        LDFLAGS=''
+        LIBS=''
+
+        # Icon-hammoz:
+        if '+ham' in self.spec:
+            args.append('--enable-atm-phy-echam-submodels')
+            args.append('--enable-hammoz')
+
+        # Eccodes library:
+        if '+eccodes' in self.spec:
+            args.append('--enable-grib2')
+            args.append('--without-external-cdi')
```

Comment:
I think the configure script uses the bundled "externals" by default. Why is this explicit switch necessary?